src/de.rs

```diff
@@ -9,17 +9,30 @@ pub fn depythonize<'de, T>(obj: &'de PyAny) -> Result<T>
 where
     T: Deserialize<'de>,
 {
-    let mut depythonizer = Depythonizer::from_object(obj);
+    let mut depythonizer = Depythonizer::from_object(obj, &None);
+    T::deserialize(&mut depythonizer)
+}
+
+/// Attempt to convert a Python object to an instance of `T`
+pub fn depythonize_on_error<'de, T>(
+    obj: &'de PyAny,
+    on_error: &'de Option<fn(&PyAny) -> &PyAny>,
```

Comment:
I haven't thought deeply about whether a different lifetime might be appropriate for `on_error`.